### PR TITLE
Add validation for visa sponsorship

### PR DESCRIPTION
**Pull request by stevehook**
### Context

Starting in the 2022 recruitment cycle we will require providers declare whether or not they can sponsor visas (student and skilled worker) before publishing any courses in that cycle. 

https://trello.com/c/I2K4fchq/3441-add-visa-flow-to-publish

### Changes proposed in this pull request

- [x] Add a new validation in the `Course` model to cover this requirement.
- [x] Add `Provider#visa_sponsorship_publishable?` as a helper method for the above validation.
- [x] Unit tests for `Provider#visa_sponsorship_publishable?` and controller spec to cover the API.

Validation as it appears in publish:
image

### Guidance to review

- Is there a better way to implement the specs for this?
- Are there any potential knock-on effects that might cause issues in the 2022 cycle?

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -119,6 +119,8 @@ en:
               blank: "^You must say when applications open from"
             base:
               duplicate: "This course already exists. You should add further locations for this course to the existing profile in Publish"
+            provider:
+              visa_sponsorship_not_publishable: "You must say whether your provider can sponsor visas"
```
Comment (davidgisbey) [inline config/locales/en.yml:123]: Should this be `You must say whether you can sponsor visas`

File: spec/models/provider_spec.rb [hunk 2]
```diff
@@ -733,4 +733,36 @@
       it { is_expected.to contain_exactly(matching_provider) }
     end
   end
+
+  describe '#visa_sponsorship_publishable?' do
+    it "returns true when year is 2022 and provider has declared sponsorship" do
```
Comment (defong) [inline spec/models/provider_spec.rb:738]: lack context 

https://www.betterspecs.org/#contexts

**Review by defong — CHANGES_REQUESTED** [review]
Has side affects
image

File: spec/controllers/api/v2/courses_controller_spec.rb [hunk 3]
```diff
@@ -35,6 +35,19 @@
         code: course.course_code,
       }
     end
+
+    it "returns a validation error when visa sponsorship information is unpublishable" do
```
[the remark this answers is not in the document]
Comment (stevehook) [reply spec/controllers/api/v2/courses_controller_spec.rb:39]: I've added a context for 2021 and 2022 - things don't change going any further forward.

**Review by davidgisbey — APPROVED** [review]
lgtm

**Review by defong — APPROVED** [review]
Let's get this moving!

**Pull request merged**